File: su/2026-05/22_pgr_기지국설치.java
class Solution {
    public int solution(int n, int[] stations, int w) {
        int answer = 0;
        int cover = 2 * w + 1; // 기지국 1개가 커버하는 범위
        int prev = 1; // 이전 기지국 커버 끝 + 1

        for (int station : stations) {
            int left = station - w; // 현재 기지국 커버 시작
            if (left > prev) {
                int gap = left - prev; // 커버 안 되는 구간 길이
                answer += (gap + cover - 1) / cover; // 올림 나눗셈
            }
            prev = station + w + 1; // 현재 기지국 커버 끝 + 1
        }

        // 마지막 기지국 이후 ~ N 구간 처리
        if (prev <= n) {
            int gap = n - prev + 1;
            answer += (gap + cover - 1) / cover;
        }

        return answer;
    }
}

File: 덕현/2026-05/22_pgr_기지국설치.py
'''
프로그래머스 12979 기지국 설치

입력:
n stations w

구조:
answer = 0 now = 1
빈 구간만 생각해서 이어지게 다음으로 넘기기
'''

def solution(n, stations, w):
    answer = 0
    now = 1
    size = 2 * w + 1


    for station in stations:
        left = station - w
        right = station + w
        if now < left:
            cnt = left - now
            answer += cnt // size
            if cnt % size != 0:
                answer += 1



        now = right + 1


    if now <= n:
        cnt = n - now + 1

        answer += cnt // size
        if cnt % size != 0:
            answer += 1

    return answer